File: Daily/PY/Leetcode1833-雪糕的最大数量.py
from typing import List

class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()
        i,ans = 0,0
        while i < len(costs) and coins >= costs[i]:
            coins -= costs[i]
            i += 1
            ans += 1
        return ans

File: Daily/Rust/Leetcode1833-雪糕的最大数量.rs
impl Solution {
    pub fn max_ice_cream(mut costs: Vec<i32>, coins: i32) -> i32 {
        costs.sort();
        let (mut ans,mut i) = (0,0);
        let mut coins = coins;
        while i < costs.len() && coins >= costs[i] {
            coins -= costs[i];
            i +=  1;
            ans += 1;
        } 
        ans
    }
}